packages/zowe-explorer/src/utils/CertificateWizard.ts

```diff
@@ -0,0 +1,127 @@
+/**
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ *
+ */
+
+import { Gui, WebView } from "@zowe/zowe-explorer-api";
+import { Disposable, ExtensionContext, OpenDialogOptions, Uri } from "vscode";
+import * as nls from "vscode-nls";
+import { ZoweLogger } from "./LoggerUtils";
+
+// Set up localization
+nls.config({
+    messageFormat: nls.MessageFormat.bundle,
+    bundleFormat: nls.BundleFormat.standalone,
+})();
+const localize: nls.LocalizeFunc = nls.loadMessageBundle();
+export type CertWizardOpts = {
+    cert?: string;
+    certKey?: string;
+    dialogOpts?: OpenDialogOptions;
+};
+
+class DeferredPromise<T> {
+    public promise: Promise<T>;
+    public resolve: (value: T | PromiseLike<T>) => void;
+    public reject: (reason?: any) => void;
+
+    public constructor() {
+        this.promise = new Promise<T>((resolve, reject) => {
+            this.resolve = resolve;
+            this.reject = reject;
+        });
+    }
+}
+
+const userDismissed = localize("zowe.certWizard.dismissed", "User dismissed the certificate wizard.");
+
+export class CertificateWizard extends WebView {
+    private onUpdateDisposable: Disposable;
+    private opts: CertWizardOpts;
+    public userSubmission: DeferredPromise<{
+        cert: string;
+        certKey: string;
+    }> = new DeferredPromise();
+
+    public constructor(context: ExtensionContext, opts: CertWizardOpts) {
+        super("Certificate Wizard", "certificate-wizard", context, (message: object) => this.onDidReceiveMessage(message));
+        this.opts = opts;
+        this.panel.onDidDispose(() => {
+            this.userSubmission.reject(userDismissed);
+        });
+    }
+
+    protected async onDidReceiveMessage(message: any): Promise<void> {
+        switch (message.command) {
+            case "promptCert":
+                {
+                    const tempCert = await Gui.showOpenDialog({
```

Comment:
This makes it super easy to follow the logic, but I'm curious if we could reduce code duplication. 
that way we have to update less things if/when we add passphrase support. 😋

Reply:
I condensed some of the dialog options passed to both `showOpenDialog` calls and removed the duplicate references - this should clean up duplication somewhat.

I'm open to other ideas of minimizing duplication here, but this should be a bit better 😋